typescript/packages/plugins/renzo/src/renzo.service.ts

```diff
@@ -0,0 +1,254 @@
+            );
+
+            const _minOut = parseUnits(parameters._minOut, 18);
+            const value = parseUnits(parameters._value, 18);
+
+            console.log(`[Renzo] Initiating ETH deposit transaction with:
+                Value: ${parameters._value} ETH
+                Min Out: ${parameters._minOut}
+                Deadline: ${parameters._deadline}`);
+
+            const deadline = BigInt(Math.floor(Date.now() / 1000) + 300); // 5 minutes from now
+
+            const hash = await walletClient.sendTransaction({
+                to: renzoDepositAddress,
+                abi: RENZO_ABI,
+                functionName: "depositETH",
+                args: [_minOut, deadline],
+                value,
+            });
+
+            console.log(
+                `[Renzo] ETH deposit transaction successful. Hash: ${hash.hash}`
+            );
+
+            return hash.hash;
+        } catch (error) {
+            console.error(`[Renzo] ETH deposit failed with error:`, error);
+            throw Error(`Failed to deposit ETH: ${error}`);
+        }
+    }
+
+    @Tool({
+        name: "renzo_check_ezeth_balance",
+        description: "Check the ezETH balance of an address",
+    })
+    async getEzEthBalance(
+        walletClient: EVMWalletClient,
+        parameters: BalanceOfParams
+    ): Promise<string> {
+        try {
+            console.log(
+                `[Renzo] Checking ezETH balance for address:`,
+                parameters._address
+            );
+
+            const { l2EzEthAddress } = getRenzoAddresses(
+                walletClient.getChain().id
+            );
+            console.log(
+                `[Renzo] Using ezETH contract address: ${l2EzEthAddress}`
+            );
+
+            const balanceResult = await walletClient.read({
+                address: l2EzEthAddress,
+                abi: EZETH_ABI,
+                functionName: "balanceOf",
+                args: [parameters._address],
+            });
+
+            console.log(`[Renzo] Balance result type:`, typeof balanceResult);
+            console.log(
+                `[Renzo] Balance result value:`,
+                JSON.stringify(balanceResult, (_, value) =>
+                    typeof value === "bigint" ? value.toString() : value
+                )
+            );
+
+            const balance = (balanceResult as { value: bigint }).value;
+            const formattedBalance = formatUnits(balance, 18);
+            console.log(
+                `[Renzo] ezETH balance for ${parameters._address}: ${formattedBalance}`
+            );
+
+            return formattedBalance;
+        } catch (error) {
+            console.error(`[Renzo] Failed to get ezETH balance:`, error);
+            throw Error(`Failed to get ezETH balance: ${error}`);
+        }
+    }
+
+    @Tool({
+        name: "renzo_approve_deposit",
+        description:
+            "Approve Renzo deposit contract to spend tokens. Call this before calling erc20 deposit function.",
```

Comment:
No need for this function, erc20 already approves. It just needs to have the token into